File: 101. Symmetric Tree.py
# -*- coding: utf-8 -*-
"""
Created on Thu Aug  9 10:12:35 2018

@author: wujiaqian
"""

# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root == None:
            return True
        return self.isSameTree(root.left,root.right)
    def isMirror(self,p,q):
        if p == None and q == None:
            return True
        elif p and q:
            if p.val == q.val:
                return self.isSameTree(p.left,q.right) and self.isSameTree(p.right,q.left)
            else:
                return False
        else:
            return False
        
            
        

File: 101.Symmetric Tree.java

public class IsSymmetric {
	public class TreeNode {
		int val;
		TreeNode left;
		TreeNode right;
		TreeNode(int x) { val = x; }
	}
	public boolean solution(TreeNode root) {
        if(root == null) {
            return true;
        }
        TreeNode a = root.left;
        TreeNode b = root.right;
        return symmetricNode(a,b);
        
    }
    public boolean symmetricNode(TreeNode a, TreeNode b) {
    	if (a == null && b == null) {
            return true;
        }
    	
        if (a.val == b.val) {
            if (a != null && b != null) {
                return(a.val == b.val && symmetricNode(a.left,b.right) && symmetricNode(a.right,b.left));
            }
            else {
                return false;
            }
        }
        else {
        	return false;
        }
    }
}
